monai/transforms/compose.py

```diff
@@ -188,21 +115,164 @@ def execute_compose(
     for _transform in transforms[start:end]:
         if threading:
             _transform = deepcopy(_transform) if isinstance(_transform, ThreadUnsafe) else _transform
-        data = evaluate_with_overrides(
-            data,
+        data = apply_transform(
             _transform,
-            lazy_evaluation=lazy_evaluation,
+            data,
+            map_items,
+            unpack_items,
+            lazy=lazy,
+            lazy_strategy=lazy_strategy,
             overrides=overrides,
-            override_keys=override_keys,
-            verbose=verbose,
+            logger_name=logger_name,
         )
-        data = apply_transform(_transform, data, map_items, unpack_items, log_stats)
-    data = evaluate_with_overrides(
-        data, None, lazy_evaluation=lazy_evaluation, overrides=overrides, override_keys=override_keys, verbose=verbose
-    )
+    if isinstance(data, (list, tuple)) and map_items:
+        data = [apply_pending_transforms(d, None, overrides, logger_name=logger_name) for d in data]
+    data = apply_pending_transforms(data, None, overrides, logger_name=logger_name)
     return data
 
 
+class ExecutionOptions:
+    """
+    ExecutionOptions is an implementation class that is required to parse options for Compose. It should currently
+    be considered an implementation detail that should not be interacted with directly by users of MONAI, although that
+    may change in subsequent releases. Its job is to parse options provided to `Compose.__call__`_ to set execution
+    modes for executing the pipeline.
+
+    See `Compose`_ for a detailed explanation of lazy resampling.
+    """
+
+    def __init__(self):
+        # construct the list of options
+        options = {"reorder": {"lazy_last": self.reorder_lazy_last, "lazy_last_nosync": self.reorder_lazy_last_nosync}}
```

Comment:
I would avoid having nested structures like this, instead:

```suggestion
        options = {("reorder","lazy_last"): self.reorder_lazy_last, ("reorder","lazy_last_nosync"): self.reorder_lazy_last_nosync}
```

Also the string keywords here should be in enums and the structure that's expected should be documented in the docstring.